fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -1119,15 +1119,31 @@ public static void stopOngoingOnlineIndexBuilds(@Nonnull FDBRecordStore recordSt
         SynchronizedSession.endAnySession(recordStore.ensureContextActive(), indexBuildLockSubspace(recordStore, index));
     }
 
-    @VisibleForTesting
-    CompletableFuture<Void> checkNoOngoingOnlineIndexBuildsAsync() {
-        return runner
-                .runAsync(context -> openRecordStore(context).thenApply(store -> indexBuildLockSubspace(store, index)))
-                .thenCompose(lockSubspace ->
-                        // It will throw {@link com.apple.foundationdb.synchronizedsession.SynchronizedSessionLockedException}
-                        // if there are ongoing online index builds.
-                        runner.startSynchronizedSessionAsync(lockSubspace, leaseLengthMills))
-                .thenCompose(SynchronizedSessionRunner::endSessionAsync);
+    /**
+     * Synchronous/blocking version of {@link #checkAnyOngoingOnlineIndexBuildsAsync()}.
+     * @return <code>true</code> if the index is being built and <code>false</code> otherwise
+     */
+    public boolean checkAnyOngoingOnlineIndexBuilds() {
+        return runner.asyncToSync(FDBStoreTimer.Waits.WAIT_CHECK_ONGOING_ONLINE_INDEX_BUILD, checkAnyOngoingOnlineIndexBuildsAsync());
+    }
+
+    /**
+     * Check if the index is being built by any of the {@link OnlineIndexer}s (only if they use {@link SynchronizedSession}s).
```

Comment:
```suggestion
     * Check if the index is being built by any other {@link OnlineIndexer}s (only if they use {@link SynchronizedSession}s).
```

I think it also might make sense to break the parenthetical into its own sentence--something like "This requires that the synchronized session feature is enabled on all indexers." (Probably with an `@see` or `@link` to the `setUseSyncrhonizedSession` method on the `OnlineIndexer.Builder`.)

Reply:
Oh, interesting. I guess that's true. I guess it's just the "the" then that's extraneous, though it may be worth calling that out in the docs (that this "self conflicts")

Reply:
Done.